ai21/clients/azure/ai21_azure_client.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from typing import Optional, Callable, Dict
+
+from ai21.ai21_http_client import AI21HTTPClient
+from ai21.clients.studio.resources.studio_chat import StudioChat
+
+AzureADTokenProvider = Callable[[], str]
+
+
+class AI21AzureClient(AI21HTTPClient):
+    def __init__(
+        self,
+        azure_endpoint: str,
+        api_version: Optional[str] = None,
+        api_key: Optional[str] = None,
+        azure_ad_token: str | None = None,
+        azure_ad_token_provider: AzureADTokenProvider | None = None,
+        default_headers: Dict[str, str] | None = None,
+        timeout_sec: int | None = None,
+        num_retries: int | None = None,
+    ):
+        self._api_key = api_key
+        self._azure_ad_token = azure_ad_token
+        self._azure_ad_token_provider = azure_ad_token_provider
+
+        if self._api_key is None and self._azure_ad_token_provider is None and self._azure_ad_token is None:
+            raise ValueError("Must provide either api_key or azure_ad_token_provider or azure_ad_token")
+
+        headers = self._prepare_headers(headers=default_headers or {})
+
+        # TODO: Change - The "/openai" suffix will probably change to "/ai21" once we have a working endpoint in Azure.
+        super().__init__(
+            api_key=api_key,
+            api_version=api_version,
+            api_host=f"{azure_endpoint}/openai",
+            headers=headers,
+            timeout_sec=timeout_sec,
+            num_retries=num_retries,
+        )
+
+        self.chat = StudioChat(self)
+        # Override the chat.create method to match the completions endpoint,
+        # so it wouldn't get to the old J2 completion endpoint
+        self.chat.create = self.chat.completions.create
+
+    def _prepare_headers(self, headers: Dict[str, str]) -> Dict[str, str]:
```

Comment:
Won't it be better if upon update we would return a new object instead of the updated one?